contracts/RiskFund/ReserveHelpers.sol

```diff
@@ -33,6 +43,40 @@ contract ReserveHelpers {
     /// @param amount An amount by which the reserves have increased
     event AssetsReservesUpdated(address indexed comptroller, address indexed asset, uint256 amount);
 
+    /// @notice event emitted on sweep token success
+    event SweepToken(address indexed token, address indexed to, uint256 amount);
+
+    /**
+     * @dev Prevents a contract from calling itself, directly or indirectly.
+     */
+    modifier nonReentrant() {
+        require(status != ENTERED, "re-entered");
+        status = ENTERED;
+        _;
+        status = NOT_ENTERED;
+    }
+
+    /**
+     * @notice A public function to sweep accidental BEP-20 transfers to this contract. Tokens are sent to the address `to`, provided in input
+     * @param _token The address of the BEP-20 token to sweep
+     * @param _to Recipient of the output tokens.
+     * @custom:error ZeroAddressNotAllowed is thrown when asset address is zero
+     * @custom:access Only Owner
+     */
+    function sweepToken(address _token, address _to) external onlyOwner nonReentrant {
+        ensureNonzeroAddress(_to);
+        uint256 balanceDfference_;
+        uint256 balance_ = IERC20Upgradeable(_token).balanceOf(address(this));
+
+        require(balance_ > assetsReserves[_token], "ReserveHelpers: Zero surplus tokens");
+        unchecked {
+            balanceDfference_ = balance_ - assetsReserves[_token];
```

Comment:
After transferring funds by mistake, if `updateAssetsState` is called either at the time of protocol share transfer / vToken or manually since it's a public function then we can not sweepTokens for the user. 
I have two suggestions:
1. We can check for the `isVToken` and also we have to implement a similar for `ProtocolShareReserve` like `isProtocolShareReserve` while calling `updateAssetsState` so that this function can only be called while transferring funds from these contracts.
2. we can pass the amount to be swept as this is an onlyOwner function and directly transfer the amount to the user without any checks.

Reply:
AFAIK tokens should be transfer to admin/owner of the contract to avoid any potential issues acc to ticket.
@chechu I think you can provide more info here

Reply:
We are already doing something similar to the first checked mentioned by @Debugger022 here: https://github.com/VenusProtocol/isolated-pools/blob/develop/contracts/RiskFund/ReserveHelpers.sol#L66

During the execution of `updateAssetsState` we check that the asset is associated with a valid market in the Comptroller. So, I don't think we need any extra check

Reply:
@chechu @GitGuru7 
Reiterating my point:

If a user mistakenly sends funds to the contract and the `updateAssetsState` function is called before executing the `sweepToken` function, the user's funds will be considered part of the reserves, and they will be unable to withdraw their funds in any way.

Reply:
I prefer to accept that case and limit the power of the owner to sweep arbitrary amounts. So, I wouldn't change it. Users shouldn't interact directly with this contract, so I hope no one really sends any tokens to it.